server/lib/src/macros.rs

```diff
@@ -486,9 +486,11 @@ macro_rules! mergemaps {
         $b:expr
     ) => {{
         $b.iter().for_each(|(k, v)| {
-            if !$a.contains_key(k) {
-                $a.insert(k.clone(), v.clone());
-            }
+            // I think to be consistent, we need the content of b to always
+            // the content of a
+            // if !$a.contains_key(k) {
+            $a.insert(k.clone(), v.clone());
+            // }
```

Comment:
This doesn't merge them in the way I'd expect, since if a contains the key, and the value's a `Vec` it *overwrites* it rather than extending, etc...?

more a `mv -R b/* a/*` than `rsync -a b/* a/`

Reply:
Yeah, think the argument order may not match command line expectations, but from the view of rust/api you can think of it like the "extend" function, which is a.extend(b), where the content of b that overlaps replaces the content of b. The same is true here, and it's how we use it, for example https://github.com/kanidm/kanidm/blob/master/server/lib/src/valueset/ssh.rs#L161 where the intent is that the content of "other" should be merged ontop of the content of self.

Merge generally actually is only used in a few places though btw. It's used for merging memberof reference sets (which are all just uuids anyway), and it's used in set_ava in entry where the existing attribute state is to be extended from an iterator/valueset.